app/models/host/managed.rb

```diff
@@ -499,10 +499,10 @@ def populate_fields_from_facts(parser, type, source_proxy)
   # The boot link and autosign entry are created
   # Any existing puppet certificates are deleted
   # Any facts are discarded
-  def setBuild
+  def setBuild(validate_on_save: true)
     self.build = true
     self.initiated_at = Time.now.utc
-    logger.warn("Set build failed: #{errors.inspect}") unless save
+    logger.warn("Set build failed: #{errors.inspect}") unless save(validate: validate_on_save)
```

Comment:
I think we should add a test that this method also works for unmanaged hosts, meaning it sets the build mode for it